src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -158,3 +164,19 @@ export const retrieveBeneficialOwners = async (req: Request, appData: Applicatio
     }
   }
 };
+
+const checkForNewlyAddedBos = (appData: ApplicationData) => {
+  const allBos = [
```

Comment:
quite expensive to concat all the lists just to check all as one

Reply:
Yeah - any suggestions? Other options I tried weren't safe as the lists are optional

Reply:
You could process each loop separately.

Reply:
Not tried this but how about something like (not actually tried this):
const hasReference = bo => bo.ch_reference === undefined;
return (appData[BeneficialOwnerIndividualKey] ?? []).find(hasReference) === true || (appData[BeneficialOwnerOtherKey] ?? []).find(hasReference) === true || (appData[BeneficialOwnerGovKey] ?? []).find(hasReference) === true;

Reply:
Tried suggestion but it gives a type mismatch between BOType/undefined and boolean.